### DISCUSSION TITLE
Is there any relation between h5 and dicom?

### DISCUSSION THREAD

**Discussion by speechle55** [facebookresearch/fastMRI#342] [Q&A]
I download the annotations from fastMRI+, but it labeled based on h5. I want to use dicom data, so is there any method to relate h5 and dicom? like how to find the dicom file for 'file1000001'

**Comment by osbm** [comment]
These files are in the special h5 format because they also have the kspace data in that file. The MRI images are called reconstructions in these files because, well, they are constructed from kspaces.

```python
import h5py
file = h5py.File(file_path, "r")
print(file.keys()) # to see what is inside this file
reconstructions_numpy = file["reconstructions_rss"][()]
```

After that you will have 3 dimensional numpy array. (num_slices, height, width)

**Reply by mahaowennn** [reply]
thx for ur reply, i know how to read h5 file, my question is how to relate the h5 file with the dicom file, because dicom can store more scanning information like 'window width' and 'orientation' so on. it seems that h5 can not store these information, and i want to use the label in h5 format to train data in dicom format. that's why i want to know how to relate h5 and dicom.

**Reply by mahaowennn** [reply]
for example, for a patient, it has both h5 format data and dicom data, and fastmti+'s labels are on h5 format, not on dicom format, so i want to know how to get the same data in both h5 and dicom format.

**Comment by mmuckley** [comment]
Hello @osbm, some of the information that you would find in the DICOM header is in the ISMRMD XML header. We have a helper function for querying the header information here: https://github.com/facebookresearch/fastMRI/blob/b66159850dbdc2569d7683f6f86bcd5cc8534339/fastmri/data/mri_data.py#L37-L69

We use this to get the final matrix resolution size. However, this only includes the information necessary for reconstruction, and other information you would find in a DICOM is not present.

To read the full ISMRMD header you'll need to read up a little bit on XML - the function above shows a few basics.
